core/src/processing/core/PShapeSVG.java

```diff
@@ -1320,17 +1333,214 @@ void setColor(String colorText, boolean isFill) {
   }
 
 
+  /**
+   * Parses the "color" datatype only, and prints an error if it is not of this form.
+   * http://www.w3.org/TR/SVG/types.html#DataTypeColor
+   * @return 0xRRGGBB (no alpha). Zero on error.
+   */
+  static protected int parseSimpleColor(String colorText) {
+    colorText = colorText.toLowerCase().trim();
+    if (colorNames.containsKey(colorText)) {
+      return colorNames.get(colorText);
+    } else if (colorText.startsWith("#")) {
+      if (colorText.length() == 4) {
+        // Short form: #ABC, transform to long form #AABBCC
+        colorText = colorText.replaceAll("^#(.)(.)(.)$", "#$1$1$2$2$3$3");
+      }
+      return (Integer.parseInt(colorText.substring(1), 16)) & 0xFFFFFF;
+      //System.out.println("hex for fill is " + PApplet.hex(fillColor));
+    } else if (colorText.startsWith("rgb")) {
+      return parseRGB(colorText);
+    } else {
+      System.err.println("Cannot parse \"" + colorText + "\".");
+      return 0;
+    }
+  }
+
+
+  /**
+   * From <a href="http://www.w3.org/TR/SVG/single-page.html#types-ColorKeywords">the spec</a>.
+   */
+  static protected HashMap<String, Integer> colorNames;
+
+  static {
```

Comment:
Is this worth 150 lines of code in this file? What application creates SVGs that uses these X Window System colors from the mid-80s?